ArcGISRuntimeSDKQt_QMLSamples/Routing/NavigateRoute/main.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <QSettings>
+#include <QGuiApplication>
+#include <QQuickView>
+#include <QCommandLineParser>
+#include <QDir>
+#include <QQmlEngine>
+
+#include "NavigateRouteSpeaker.h"
+
+static QObject *example_qobject_singletontype_provider(QQmlEngine *engine, QJSEngine *scriptEngine)
+{
+    Q_UNUSED(engine)
+    Q_UNUSED(scriptEngine)
+
+    NavigateRouteSpeaker *example = new NavigateRouteSpeaker();
+    return example;
+}
+
+int main(int argc, char *argv[])
+{
+  QGuiApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
+  QGuiApplication app(argc, argv);
+  app.setApplicationName(QStringLiteral("NavigateRoute - QML"));
+
+  // Intialize application view
+  QQuickView view;
+  view.setResizeMode(QQuickView::SizeRootObjectToView);
+
+  // Register the C++ NavigateRouteSpeaker class
+  qmlRegisterSingletonType<NavigateRouteSpeaker>("Esri.samples", 1, 0, "NavigateRouteSpeaker", [](QQmlEngine* parent, QJSEngine*)-> QObject* { return new NavigateRouteSpeaker
+        (parent);  } );
+
+  // Add the import Path
+  view.engine()->addImportPath(QDir(QCoreApplication::applicationDirPath()).filePath("qml"));
+
+#ifdef ARCGIS_RUNTIME_IMPORT_PATH_2
+  view.engine()->addImportPath(ARCGIS_RUNTIME_IMPORT_PATH_2);
+#endif
+
+#ifdef ARCGIS_TOOLKIT_IMPORT_PATH_2
+  view.engine()->addImportPath(ARCGIS_TOOLKIT_IMPORT_PATH_2);
+#endif
```

Comment:
Why the idefs? Is this standard practice? Took a glance at a couple of the other samples and they don't seem to have them.

Reply:
This just came as part of the sample template

Reply:
Fair enough, suppose it can't hurt.